pkg/orchestration/wiring/registry/registry.go

```diff
@@ -16,19 +16,31 @@ import (
 	"github.com/atlassian/voyager/pkg/orchestration/wiring/ups"
 	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin"
 	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/oap"
 )
 
-var KnownWiringPlugins = map[voyager.ResourceType]wiringplugin.WiringPlugin{
-	apik8scompute.ResourceType:  wiringutil.StatusAdapter(k8scompute.WireUp),
-	kubeingress.ResourceType:    wiringutil.StatusAdapter(kubeingress.WireUp),
-	ec2compute_v2.ResourceType:  wiringutil.StatusAdapter(ec2compute_v2.WireUp),
-	ups.ResourceType:            wiringutil.StatusAdapter(ups.New().WireUp),
-	aws.Cfn:                     aws.ResourceTypes[aws.Cfn],
-	aws.DynamoDB:                aws.ResourceTypes[aws.DynamoDB],
-	aws.S3:                      aws.ResourceTypes[aws.S3],
-	postgres.ResourceType:       wiringutil.StatusAdapter(postgres.New().WireUp),
-	rds.ResourceType:            wiringutil.StatusAdapter(rds.New().WireUp),
-	sqs.ResourceType:            wiringutil.StatusAdapter(sqs.WireUp),
-	asapkey.ResourceType:        wiringutil.StatusAdapter(asapkey.New().WireUp),
-	apiplatformdns.ResourceType: wiringutil.StatusAdapter(platformdns.New().WireUp),
+func KnownWiringPlugins(
```

Comment:
Sort of related to Nail's comment up https://github.com/atlassian/voyager/pull/163/files#r256667823, we only need to do this because we need to pass a bunch of parameters through the orchestration constructor. When we end up extracting these I assume we'll have a custom orchestration cmd in stash voyager that combines the wiring (like how we combine smith plugins and smith)

Reply:
I have some ideas of how we can have configuration passed to plugins from a shared config file but in a typed way. I view passing functions is a temporary shortcut - instantiating code (orchstration) should not have any knowledge about what configuration is needed i.e. that "configuration" cannot be a function. It should be just a blob of data. But this is one of the next steps.

Reply:
And same with smith - we need a way to pass configuration to plugins https://github.com/atlassian/smith/issues/319 without recompiling smith to change that configuration. We obviously don't want to have a binary per environment, for example.